### ISSUE TITLE
filter is reset on changing color, size, x, y

### ISSUE THREAD

**Issue by abast** [flekschas/jupyter-scatter#134]
Create scatter plot with random data:

```
data = pd.DataFrame({
    "x": np.random.rand(1000),
    "y": np.random.rand(1000),
    "z": np.random.rand(1000),
})

scatter = Scatter(data=data, x="x", y="y")
scatter.filter()
scatter.show()
```

Show only points for which x > 0.5
```
indices = list(data[data['x']>0.5].index)
scatter.filter(indices)
```

Each of the following lines resets the selection:
```
scatter.color(by = 'z')
scatter.size(by = 'z')
scatter.x('y')
scatter.y('x')
```

**Comment by flekschas** [comment]
Thanks for reporting the issue. This seems to be a bug in regl-scatterplot. I'm investigating

**Comment by flekschas** [comment]
Fixed in https://github.com/flekschas/jupyter-scatter/commit/5dbceffb6e7533c15ee3e6b8ec1047372625d156 and released as v0.16.1.



https://github.com/flekschas/jupyter-scatter/assets/932103/50e6795c-c16f-4b24-a9d9-39bde4f477fe
